book/modules/shared/partials/common-nav.adoc

```diff
@@ -1 +1 @@
-* xref:shared:common-page.adoc[Link to common page]
+* xref:{page-module}:shared/common-page.adoc[Link to common page]
```

Comment:
Using the in-built `page-module` attribute here will make sure it is evaluated in the context of the module including this nav. That means it will point to the symlinked `modules/mod-*/pages/shared/...` pages instead of pointing to `modules/shared/pages/shared/...`. This completes the trick